otherlibs/systhreads/st_posix.h

```diff
@@ -426,3 +432,48 @@ value caml_wait_signal(value sigs) /* ML */
   return Val_int(0);            /* not reached */
 #endif
 }
+
+#include "../../config/s.h"
+
+void st_determine_thread_stack_size(void** top_of_stack, size_t* stack_size)
+{
+  /* Note: at least under NPTL, the returned size will include the guard
+     page size. */
+
+  int local_var;
+
+  *top_of_stack = (void*) &local_var;
+  *stack_size = -1;
+
+#if defined(HAS_PTHREAD_GET_STACKSIZE_NP) \
```

Comment:
This #if nest is a pain to read because there is no consistent return style: sometimes `return` means failure and sometimes `return` means success.  I suggest to stick to `return` mean success and fallthrough means keep trying:
```
  if (PTHREAD_GETATTR_NP(pthread_self(), &attr) == 0 &&
      pthread_attr_getstack(&attr, top_of_stack, stack_size) == 0) {
   *top_of_stack += *stack_size;
    return; /* Success! */
  }
```
This works even if several #if feature tests succeed, and does the right thing: the first feature is tried, if it succeeds we return, otherwise we fall through, and the second feature is tried.  Finally, the default can be at the end of the function.